pkg/api/types.go

```diff
@@ -1022,15 +925,16 @@ func (p *Properties) GetPrimaryScaleSetName() string {
 
 // IsIPMasqAgentEnabled returns true if ip-masq-agent is enabled
 func (p *Properties) IsIPMasqAgentEnabled() bool {
-	return p.OrchestratorProfile.KubernetesConfig.IsIPMasqAgentEnabled()
+	return p.OrchestratorProfile != nil &&
+		p.OrchestratorProfile.KubernetesConfig != nil &&
+		p.OrchestratorProfile.KubernetesConfig.IsAddonEnabled(common.IPMASQAgentAddonName)
 }
 
 // IsIPMasqAgentDisabled returns true if the ip-masq-agent functionality is disabled
 func (p *Properties) IsIPMasqAgentDisabled() bool {
-	if p.OrchestratorProfile != nil && p.OrchestratorProfile.KubernetesConfig != nil {
-		return p.OrchestratorProfile.KubernetesConfig.IsIPMasqAgentDisabled()
-	}
-	return false
+	return p.OrchestratorProfile != nil &&
```

Comment:
Alternate implementation: `return !p.IsIPMasqAgentEnabled()`.

Reply:
so this is a confusing one that I stumbled around as well

Basically there's a race condition between kubelet and addons defaults enforcement, and they depend on each other for this particular config. So what we want to confirm is that the addon is *explicitly* disabled. Whereas the "is enabled" foo will return false when it's empty/nil/whatever, which is misleading, because if you consider that authoritative and then the addons default foo sets it to true later in the code flow, you're in trouble.

Luckily we have UT to tell this story, and when I did the obvious thing that you suggested they complained!

Reply:
Ah, I was wondering if it was more of a tri-state boolean but I made the naive comment anyway. I don't remember these details for very long apparently. :-)

Reply:
Also all hail the UT.